src/crimson/osd/pg.h

```diff
@@ -174,51 +179,85 @@ class PG : public boost::intrusive_ref_counter<
     // will be needed for unblocking IO operations/peering
   }
 
+  template <typename T>
+  void start_peering_event_operation(T &&evt) {
+    shard_services.start_operation<LocalPeeringEvent>(
+      this,
+      shard_services,
+      pg_whoami,
+      pgid,
+      std::forward<T>(evt));
+  }
+
   void schedule_event_after(
     PGPeeringEventRef event,
     float delay) final {
-    ceph_assert(0 == "Not implemented yet");
+    // TODO: this is kind of a hack -- once the start_operation call
+    // happens, the operation will be registered, but during the delay
+    // it's just a dangling future.  It would be nice for the
+    // operation machinery to have something to take care of this.
+    (void)seastar::sleep(std::chrono::milliseconds(std::lround(delay*1000))).then(
```

Comment:
I'm ok with leaving this for now, but we need to address it in part 2.  I'd think it would be a simple as adding a blocking future wrapping a sleep?

Reply:
Sorry, it appears that I overlooked this comment. I've fix it, please take a look, thanks:-)